integration-test/src/test/java/org/jboss/pnc/integration/endpoints/BuildEndpointTest.java

```diff
@@ -731,6 +732,21 @@ public void shouldNotApplyUnknownQualityLevel() throws RemoteResourceException {
                 .hasCauseInstanceOf(BadRequestException.class);
     }
 
+    @Test
+    @InSequence(0)
+    public void shouldReportLeafTempBuildThatDependsOnItself() throws RemoteResourceException {
+        BuildClient client = new BuildClient(RestClientConfiguration.asUser());
+
+        Collection<Build> leafTempBuilds = client.getAllIndependentTempBuildsOlderThanTimestamp(Long.MAX_VALUE)
+                .getAll();
+
+        assertThat(leafTempBuilds).hasSize(1);
+        Map<String, String> attributes = leafTempBuilds.iterator().next().getAttributes();
+        assertThat(attributes)
+                .containsEntry(org.jboss.pnc.api.constants.Attributes.BUILD_BREW_NAME, "org.jboss.pnc:parent");
+        assertThat(attributes).containsEntry(org.jboss.pnc.api.constants.Attributes.BUILD_BREW_VERSION, "1.2.4");
```

Comment:
Are we sure we have the right one here? org.jboss.pnc:parent:1.2.4 is twice in the demo data - one persistent and one temporary.

Reply:
it should return only temporary, but I'll add chceck that it is the temporary one